Video/graysonsolis_Auto change video.lua

```diff
@@ -0,0 +1,124 @@
+-- @description Auto change video
+-- @author Grayson Solis
+-- @version 1.0
+-- @about
+--   - Auto selects nearest video track
+--
+--   - If selected track has video, unmute track’s video items and mute all others
+--   - If selected track has no video, finds nearest track above it with video and unmutes that video item instead
+--   - Ignores muted or collapsed tracks
+-- @website https://graysonsolis.com
+-- @donations https://paypal.me/GrayTunes?country.x=US&locale.x=en_US
+
+-- Caching for performance
+
+local GetTrack              = reaper.GetTrack
+local GetParentTrack        = reaper.GetParentTrack
+local GetMediaTrackInfo     = reaper.GetMediaTrackInfo_Value
+local CountTracks           = reaper.CountTracks
+local CountMediaItems       = reaper.CountTrackMediaItems
+local GetMediaItem          = reaper.GetTrackMediaItem
+local GetTake               = reaper.GetActiveTake
+local GetSource             = reaper.GetMediaItemTake_Source
+local GetSourceType         = reaper.GetMediaSourceType
+local SetItemMute           = reaper.SetMediaItemInfo_Value
+local PreventUIRefresh      = reaper.PreventUIRefresh
+local defer                 = reaper.defer
+local time_precise          = reaper.time_precise
+local GetSelectedTrack      = reaper.GetSelectedTrack
+local CountSelectedTracks   = reaper.CountSelectedTracks
+
+local function isVisible(tr)
+  while tr do
+    tr = GetParentTrack(tr)
+    if tr and GetMediaTrackInfo(tr, "I_FOLDERCOMPACT") == 1 then 
```

Comment:
This only checks for the first collapsed level (I_FOLDERCOMPACT==1) but ignores the second "fully collapsed" level (I_FOLDERCOMPACT==2) – is that the correct behavior?